Packs/CommonScripts/Scripts/FileEnrichment/FileEnrichment_test.py

```diff
@@ -0,0 +1,606 @@
+    """
+    command = Command("file", {"file": SHA_256_HASH})
+
+    demisto_execute_response = util_load_json("test_data/file_reputation_command_response.json")
+    mock_demisto_execute = mocker.patch("FileEnrichment.demisto.executeCommand", return_value=demisto_execute_response)
+
+    entry_context, readable_command_results = command.execute()
+
+    assert mock_demisto_execute.call_count == 1
+    assert entry_context[0] == demisto_execute_response[0]["EntryContext"]
+    assert readable_command_results[0].readable_output == (
+        f"#### Result for {command}\n{demisto_execute_response[0]['HumanReadable']}"
+    )
+
+
+def test_get_file_from_ioc_custom_fields():
+    """
+    Given:
+        - A File indicator with `CustomFields`.
+
+    When:
+        - Calling `get_file_from_ioc_custom_fields`.
+
+    Assert:
+        - Ensure correct `File` context that removes empty values and extracts `Name` and `Signature`.
+    """
+    from FileEnrichment import get_file_from_ioc_custom_fields
+
+    ioc_custom_fields = {
+        "fileextension": "ZIP",
+        "filetype": "application/zip",
+        "md5": "4E76823C05048E92A4C0122D61000EDF",
+        "sha1": "D8B426700C3C10413ABB8ACDCFECCAAEC8F06CD9",
+        "sha256": "7AA15BD505A240A8BF62735A5389A530322945EEC6CE9D7B6AD299CA33B2B1B0",
+        "size": 262291,
+        "ssdeep": "6144:WWEH6PCEYPX3DNGG+KFXLLRNR8M+FHAD2HNRG/U/:WWS6PCVLHT+KDR8/AD23Z",
+        "tags": ["checks-cpu-name", "checks-hostname"],
+        "stixid": None,  # expect to be removed (empty value)
+        "associatedfilenames": ["VS Code", "Visual Studio Code Setup (x64)"],
+        "signatureauthentihash": "7AA15BD505A240A8BF62735A5389A530322945EEC6CE9D7B6AD299CA33B2B1B0",
+    }
+    file_indicator_context = get_file_from_ioc_custom_fields(ioc_custom_fields)
+
+    assert "stixid" not in file_indicator_context
+    assert file_indicator_context == {
+        "Extension": ioc_custom_fields["fileextension"],
+        "Type": ioc_custom_fields["filetype"],
+        "MD5": ioc_custom_fields["md5"],
+        "SHA1": ioc_custom_fields["sha1"],
+        "SHA256": ioc_custom_fields["sha256"],
+        "Size": ioc_custom_fields["size"],
+        "SSDeep": ioc_custom_fields["ssdeep"],
+        "Tags": ioc_custom_fields["tags"],
+        "AssociatedFileNames": ioc_custom_fields["associatedfilenames"],
+        "Name": ioc_custom_fields["associatedfilenames"][0],
+        "Signature": {
+            "Authentihash": ioc_custom_fields["signatureauthentihash"],
+        }
+    }
+
+
+def test_flatten_list():
+    """
+    Given:
+        - A list of numbers that contains a nested list of numbers.
+
+    When:
+        - Calling `flatten_list`.
+
+    Assert:
+        - Ensure a flattened (non-nested) list of the same numbers.
+    """
+    from FileEnrichment import flatten_list
+
+    nested_list = ["1", "2", ["3", "4"], "5"]
```

Comment:
what would happen in this case- [1,2,3,[4,5],4,6]?

Reply:
Updated logic to handle a wider range of data types